docs/api/effector/sample.md

```diff
@@ -0,0 +1,197 @@
+const count = createStore(0)
+  .on(inc, (state) => state + 1);
+
+const sampled = sample(event, count, (c, i) => `Current count is ${i}, last event invocation: ${c}`);
+sampled.watch(console.log);
+
+inc() // => nothing
+
+event("foo");
+inc() // => Current count is 2, last event invocation: foo 
+
+event("bar");
+inc(); // => Current count is 3, last event invocation: bar 
+
+```
+
+# `sample(sourceStore, clockStore, fn)`
+
+Passes last `sourceStore`'s current state and `clockStore`'s updated state to `fn` handler, upon `clockStore`'s update.
+
+#### Arguments
+
+1. `sourceStore` _(Store)_: Source store
+2. `clockStore` _(Store)_: Triggers sampled unit upon store update
+3. `fn`? _(Function)_: Callable handler (may be [_`Event`_](Event.md) or [_`Effect`_](Effect.md) aswell), should be **pure**
+
+#### Returns
+
+([_`Store`_](Store.md)) - Store, which updates upon clock update
+
+#### Example
+
+```js
+const inc = createEvent();
+const setName = createEvent();
+
+const name = createStore("John")
+  .on(setName, (_, v) => v);
+
+const clock = createStore(0)
+  .on(inc, (i) => i + 1);
+
+const sampled = sample(name, clock, (name, i) => `${name} has ${i} coins`);
+sampled.watch(console.log);
+ // => John has 0 coins (initial store update triggered sampled store)
+
+setName("Doe");
+inc(); // => Doe has 1 coins 
+
+```
+
+# `sample({source, clock, fn, greedy?})`
+
+Object-like arguments passing, working exactly the same sa examples above.
+
+`greedy` modifier defines, whether sampler will wait of  resolving calculation result, and will batch all updates, resulting only one trigger, either will be triggered upon every linked node invocation, e.g. if `greedy` is `true`, `sampler` will fire, upon trigger of every node, linked to clock, whereas `non-greedy sampler(greedy: false)` will fire upon the last linked node trigger.
+
+#### Arguments
+
+1. `params` _(Object)_: Configuration object
+
+#### Returns
+
+([_`Event`_](Event.md)|[_`Store`_](Store.md)) - Unit, which fires/updates upon `clock` is trigged
+
+#### Example
+
+```js
+const lastEvent = createStore(null)
+  .on(clickButton, (_, data) => data)
+  .on(closeModal, () => 'modal')
+
+lastEvent.updates.watch(data => {
+  //here we need everything
+  //console.log(`sending important analytics event: ${data}`)
+})
+
+lastEvent.updates.watch(data => {
+  //here we need only final value
+  //console.log(`render <div class="yourstatus">${data}</div>`)
+})
+
+const analyticReportsEnabled = createStore(false)
+
+const commonSampling = sample({
+  source: analyticReportsEnabled,
+  clock: merge([clickButton, closeModal]),
+  fn: (isEnabled, data) => ({isEnabled, data}),
+})
+
+const greedySampling = sample({
+  source: analyticReportsEnabled,
+  clock: merge([clickButton, closeModal]),
+  fn: (isEnabled, data) => ({isEnabled, data}),
+  greedy: true,
+})
+
+commonSampling.watch(data => console.log('non greedy update', data))
+greedySampling.watch(data => console.log('greedy update', data))
+
```

Comment:
Can we provide links to an online editor for examples? Some users read the docs in markdown.